ui/src/components/admin/ModelsManager/ModelManagerController.js

```diff
@@ -115,350 +120,356 @@ Model table state transitions for each action:
+                      type: NOTIFICATION_TYPES.LOADING,
+                      message: 'Initiating Bulk Publish...',
+                      details:
+                        'You can continue to use the CMS and will be notified when the publish is complete. Please note that some functions are unavailable during the publish process.',
                       timeout: false,
                     });
-                  });
-              },
-              bulkPublish: bulkActionCreator({
-                action: 'publish',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              bulkUnpublish: bulkActionCreator({
-                action: 'unpublish',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              bulkDelete: bulkActionCreator({
-                action: 'delete',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              publishOne: async name => {
-                // Set loading true (lock UI)
-                await setState({
-                  isLoading: true,
-                });
-
-                singleAction('publish', name)
-                  .then(() => loadData(baseUrl, state))
-                  .then(async ([dataResponse, countResponse]) => {
+                    await publishBulk(state.selection)
+                      .then(async response => {
+                        if (response.data.success) {
+                          await setPublishProgress({
+                            ...publishProgress,
+                            running: true,
+                          });
+                          await tempNotification.clear();
+                        }
+                      })
+                      .catch(async error => {
+                        await appendNotification({
+                          type: NOTIFICATION_TYPES.ERROR,
+                          message: 'Bulk Publish Failed.',
+                          details: error.response
+                            ? error.response.data.error.message
+                            : error.message,
+                          timeout: false,
+                        });
+                        await tempNotification.clear();
+                      });
+                  },
+                  bulkUnpublish: bulkActionCreator({
+                    action: 'unpublish',
+                    baseUrl,
+                    state,
+                    setState,
+                    appendNotification,
+                  }),
+                  bulkDelete: bulkActionCreator({
+                    action: 'delete',
+                    baseUrl,
+                    state,
+                    setState,
+                    appendNotification,
+                  }),
+                  publishOne: async name => {
```

Comment:
`publishOne` (individual publish from Model Management table) has also been updated. It no longer locks the UI with `isLoading`, and instead calls the async publish function and creates an individual publish notification for that model.